xpublish/rest.py

```diff
@@ -97,18 +100,26 @@ def zmetadata_json(self):
         return zjson
 
     def get_key(self, var, chunk):
+        cache_key = f'{var}/{chunk}'
         logger.debug('var is %s', var)
         logger.debug('chunk is %s', chunk)
 
-        da = self._variables[var].data
-        arr_meta = self.zmetadata['metadata'][f'{var}/{array_meta_key}']
+        response = self._cache.get(cache_key)
+        if response is None:
+            with CostTimer() as ct:
+                arr_meta = self.zmetadata['metadata'][f'{var}/{array_meta_key}']
+                da = self._variables[var].data
 
-        data_chunk = get_data_chunk(da, chunk, out_shape=arr_meta['chunks'])
+                data_chunk = get_data_chunk(da, chunk, out_shape=arr_meta['chunks'])
 
-        echunk = _encode_chunk(
-            data_chunk.tobytes(), filters=arr_meta['filters'], compressor=arr_meta['compressor'],
-        )
-        return Response(echunk, media_type='application/octet-stream')
+                echunk = _encode_chunk(
+                    data_chunk.tobytes(),
+                    filters=arr_meta['filters'],
+                    compressor=arr_meta['compressor'],
+                )
+                response = Response(echunk, media_type='application/octet-stream')
+            self._cache.put(cache_key, response, max(1, int(ct.time)), len(echunk))
```

Comment:
This is neat! So, is this being cached in the same memory pool used by the server?

Reply:
Yes, exactly! Which I think is what we need anyways.